src/Json.Schema.ToDotNet.UnitTests/Hints/ClassNameHintTests.cs

```diff
@@ -269,7 +269,208 @@ public int GetHashCode(FileData obj)
         }
     }
 }"
-)
+                ),
+
+            new TestCase(
+                "Renamed class has a base class",
+@"{
+  ""type"": ""object"",
+  ""properties"": {
+    ""file"": {
+      ""$ref"": ""#/definitions/file""
+    }
+  },
+  ""definitions"": {
+    ""file"": {
+      ""type"": ""object"",
+      ""properties"": {
+        ""path"": {
+          ""type"": ""string""
+        }
+      }
+    }
+  }
+}",
```

Comment:
The schema defines a root object with a single property named `file` of type `file`.